Documentation/Patterns.md

```diff
@@ -0,0 +1,85 @@
+# SerenityOS patterns
+
+----
```

Comment:
not sure why we need an `<hr>` before each section?

Reply:
Following the pattern that is is in `Documentation/SmartPointers.md`? I have no preference.

Reply:
Removed since I don't really care either way.